### PR TITLE
fix(gh-action): do not fetch branch

### PR DESCRIPTION
**Pull request by jodeppo**
## Summary:

This Github action fetched the branch name before triggering. I don't think this is needed since you can use Github ref's, like shown in https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request where it says:
image

Github holds those special refs to easily link to each PR.

Additionally, I've seen this action failing, e.g. here: https://github.com/facebook/react-native/actions/runs/11698369279/job/32578542816#step:4:41

## Changelog:

[Internal] [Fixed] - Use `refs/pulls` namespace in trigger action

### CODE REVIEW

**Review by cortinico — APPROVED** [review]

File: .github/workflows/trigger-e2e-on-comment.yml [hunk 1]
```diff
@@ -22,23 +22,16 @@ jobs:
       - name: Run E2E Tests
```
Comment (cortinico) [inline .github/workflows/trigger-e2e-on-comment.yml:22]: Can you remove the `brew install jq`?

File: .github/workflows/trigger-e2e-on-comment.yml [hunk 2]
```diff
@@ -22,23 +22,16 @@ jobs:
...
+          echo "Trigger Test All workflow for PR $PR_NUMBER"
+          
           curl -L \
             -X POST \
             -H "Accept: application/vnd.github+json" \
             -H "Authorization: Bearer $GITHUB_TOKEN" \
             -H "X-GitHub-Api-Version: 2022-11-28" \
             https://api.github.com/repos/facebook/react-native/actions/workflows/test-all.yml/dispatches \
-            -d "{\"ref\": \"$BRANCH\", \"inputs\": {\"run-e2e-tests\": \"true\"}}"
+            -d "{\"ref\": \"refs/pull/$PR_NUMBER/merge\", \"inputs\": {\"run-e2e-tests\": \"true\"}}"
```
Comment (cortinico) [inline .github/workflows/trigger-e2e-on-comment.yml:34]: Can't get just use `GITHUB_REF` here instead?

**Review by cortinico — CHANGES_REQUESTED** [review]

File: .github/workflows/trigger-e2e-on-comment.yml (hunk 2 above)
Comment (jodeppo) [reply .github/workflows/trigger-e2e-on-comment.yml:34]: Don't think so. `GITHUB_REF` at `issue_comment` events will use `default branch`, but this trigger should use the branch of the PR the comment was added to.

image
https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#issue_comment

File: .github/workflows/trigger-e2e-on-comment.yml (hunk 1 above)
Comment (jodeppo) [reply .github/workflows/trigger-e2e-on-comment.yml:20]: yep, done

**Comment by cipolleschi** [comment]
Hi @jodeppo, thanks for trying and fixing this, but the problem here is that this action is triggered by a `comment`, not by a commit.
Given the comment, the base branch used, the `GITHUB_REF`, is `main`, not the branch used by the PR itself. That's why the need of fetching the name.

> Additionally, I've seen this action failing, e.g. here: [facebook/react-native/actions/runs/11698369279/job/32578542816#step:4:41](https://github.com/facebook/react-native/actions/runs/11698369279/job/32578542816#step:4:41)

The Action is also failing because it doesn't work on forks. It work properly only on branches on the main repo.

---

The action has been created to simplify some testing for an umbrella task that is closed now.
I believe that we can simply remove the action.

**Comment by jodeppo** [comment]
> Hi @jodeppo, thanks for trying and fixing this, but the problem here is that this action is triggered by a `comment`, not by a commit. Given the comment, the base branch used, the `GITHUB_REF`, is `main`, not the branch used by the PR itself. That's why the need of fetching the name.

Ah, gotcha. Sorry - my comment about `GITHUB_REF` didn't really make sense. The change itself should probably still work, since it's just using the `PR_NUMBER`, which uses `${{ github.event.issue.number }}`, not `GITHUB_REF`

**Comment by cipolleschi** [comment]
@jodeppo cool. Given my comment: 

> The action has been created to simplify some testing for an umbrella task that is closed now.
I believe that we can simply remove the action.

Would you mind changing the PR to just remove the action, instead?

**Comment by jodeppo** [comment]
done

**Pull request closed without merging**